gossip/gossip.go

```diff
@@ -251,17 +243,51 @@ func (g *Gossip) AddInfo(key string, val interface{}, ttl time.Duration) error {
 	return err
 }
 
+// AddInfoProto adds or updates an info object. Returns an error if info
+// couldn't be added.
+func (g *Gossip) AddInfoProto(key string, proto gogoproto.Message, ttl time.Duration) error {
+	bytes, err := gogoproto.Marshal(proto)
+	if err != nil {
+		return err
+	}
+	return g.AddInfo(key, bytes, ttl)
+}
+
 // GetInfo returns an info value by key or an error if specified
 // key does not exist or has expired.
-func (g *Gossip) GetInfo(key string) (interface{}, error) {
+func (g *Gossip) GetInfo(key string) ([]byte, error) {
 	g.mu.Lock()
-	defer g.mu.Unlock()
-	if i := g.is.getInfo(key); i != nil {
-		return i.Val, nil
+	i := g.is.getInfo(key)
+	g.mu.Unlock()
```

Comment:
I think the more common pattern is just using defer. That pattern is preferred because using defer to unlock resources is less fragile in more complicated situations. We'd certainly prefer defer if there was some kind of `if .. { return err }` between the locking and unlocking. That's a pretty strong argument to just use defer.